public/app/core/services/bridge_srv.ts

```diff
@@ -62,6 +65,18 @@ export class BridgeSrv {
         });
         console.log('store updating angular $location.url', url);
       }
+
+      // Check for template variable changes
+      if (this.lastQuery) {
```

Comment:
think we also need to check if we are still on the same dashboard. This location change could changing dashboard  (to another dashboard with variables in the url)